src/core.jl

```diff
@@ -0,0 +1,137 @@
+import Base: push!, eltype, close
+export Signal, Input, Node, push!, value, close
+
+##### Node #####
+
+immutable Action
+    recipient::WeakRef
+    f::Function
+end
+isrequired(a::Action) = a.recipient.value != nothing && a.recipient.value.alive
+
+type Node{T}
+    value::T
+    actions::Vector{Action}
+    alive::Bool
+end
```

Comment:
Yeah, a field makes sense. It's probably not worth the extra compilation overhead to do it with a new type or type parameter.